### PR TITLE
Cancelling.

### PR DESCRIPTION
**Pull request by eightypop**
Addresses https://github.com/niksy/throttle-debounce/issues/6.

I started this a few days ago but I see there's already kind of an implementation in the time it took me to create this one. I figured I might as well open the PR for at least some discussion. 

I think this impl may be a little more thorough since after cancelling the the callback function will not be called again regardless of whether or not it was going to be executed in a timeout or synchronously. 

This also moves the documentation to the Usage section and provides an example.

### CODE REVIEW

**Comment by niksy** [comment]
This looks great! Maybe you can combine `clearTimeout` calls into one method?

File: README.md [hunk 1]
```diff
@@ -25,6 +25,18 @@ debounce(300, function () {
 	// Debounced function
 });
 ```
+### cancelling
+
+Debounce and throttle can both be cancelling by calling the `cancel` function.
```
Comment (niksy) [inline README.md]: `cancelling` => `cancelled`

File: README.md [hunk 2]
```diff
@@ -25,6 +25,18 @@ debounce(300, function () {
...
+
+```js
+ var throttled = throttle(300, function () {
```
Comment (niksy) [inline README.md]: Could you please change `var` into `const` and function declaration into arrow function? I’m trying to promote modern-ish approach in docs :) You’re also welcome to change other references in README!

**Comment by eightypop** [comment]
@niksy What do you think about merging this?

**Comment by CGNonofr** [comment]
Are we waiting for something before merging and releasing this?

**Comment by ATypescriptEnjoyer** [comment]
Just found this task trying to figure out how to cancel a debounced function, is there an ETA on this being deployed? :+1:

**Review by niksy — CHANGES_REQUESTED** [review]

**Comment by niksy** [comment]
@whatknight sorry for the wait! Can you please look at the requested changes?

**Comment by eightypop** [comment]
@niksy thanks for the comments! i made the requested changes to the readme. let me know if there's anything else we need to do before we merge.

**Review by niksy — APPROVED** [review]

**Comment by niksy** [comment]
@whatknight could you please rebase with `master` so we can merge your changes?

**Comment by eightypop** [comment]
@niksy Just updated, sorry it took so long 😞 , been in a crunch.

**Pull request merged**